src/realm/object-store/c_api/query.cpp

```diff
@@ -194,24 +195,39 @@ RLM_API realm_query_t* realm_query_parse(const realm_t* realm, realm_class_key_t
 {
     return wrap_err([&]() {
         auto table = (*realm)->read_group().get_table(TableKey(target_table_key));
-        DescriptorOrdering ordering;
+        util::bind_ptr<DescriptorOrdering> ordering;
         Query query = parse_and_apply_query(*realm, table, ordering, query_string, num_args, args);
         return new realm_query_t{std::move(query), std::move(ordering), *realm};
     });
 }
 
+RLM_API const char* realm_query_get_description(realm_query_t* query)
+{
+    return wrap_err([&]() {
+        query->query.set_ordering(query->ordering);
+        auto cleanup = util::ScopeExit([query]() noexcept {
+            query->ordering = query->query.get_ordering();
+        });
+        const char* str = query->query.get_description().c_str();
```

Comment:
Looking at how this gets changed below, can we move the owned string to `realm_query_t` rather than holding a std::string in side `Query`?

Reply:
Sure - don't know why I did not get that idea.